src/OrchardCore.Modules/OrchardCore.Liquid/Endpoints/Api/GetIntellisenseEndpoint.cs

```diff
@@ -0,0 +1,74 @@
+using Fluid;
+using System.Text;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Microsoft.Net.Http.Headers;
+using OrchardCore.ContentManagement;
+using OrchardCore.DisplayManagement.Liquid;
+using OrchardCore.Environment.Shell.Configuration;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace OrchardCore.Liquid.Endpoints.Api;
+
+public static class GetIntellisenseEndpoint
+{
+
+    public static IEndpointRouteBuilder AddGetIntellisenseEndpoint(this IEndpointRouteBuilder builder)
+    {
+        builder.MapGet("/OrchardCore.Liquid/Scripts/liquid-intellisense.js", HandleAsync)
+            .AllowAnonymous()
+            .DisableAntiforgery();
+
+        return builder;
+    }
+
+    [Authorize(AuthenticationSchemes = "Api")]
+    private static Task<IResult> HandleAsync(HttpContext httpContext, IMemoryCache cache)
+    {
+        const string cacheKey = "LiquidIntellisenseScript";
+
+        var cacheControl = $"public, max-age={TimeSpan.FromDays(30).TotalSeconds}, s-max-age={TimeSpan.FromDays(365.25).TotalSeconds}";
```

Comment:
Can the etag and caching headers be configured on the `MapGet` ? So we don't have to handle the checks ourselves?